collector/agent_poller.go

```diff
@@ -69,80 +61,120 @@ type AgentState struct {
 }
 
 // ---
 
-func pollAgent(agentIp string, recordsChan chan<- *AvroDatum, agentStateChan chan<- *AgentState, stats chan<- StatsEvent) {
+func (a *Agent) pollAgent(recordsChan chan<- *AvroDatum, stats chan<- StatsEvent) {
 	// always fetch/emit agent state first: downstream will use it for tagging metrics
-	agentState, err := getAgentState(agentIp, stats)
+	agentState, err := a.getAgentState(stats)
 	if err == nil {
-		agentStateChan <- agentState
+		a.AgentStateChan <- agentState
 	} else {
-		log.Printf("Failed to retrieve state from agent at %s: %s", agentIp, err)
+		log.Printf("Failed to retrieve state from agent at %s: %s", a.AgentIP, err)
 	}
 
-	systemMetricsList, err := getSystemMetrics(agentIp, agentState, stats)
+	systemMetricsList, err := a.getSystemMetrics(agentState, stats)
 	if err == nil {
 		if systemMetricsList != nil {
 			recordsChan <- systemMetricsList
 		}
 	} else {
-		log.Printf("Failed to retrieve system metrics from agent at %s: %s", agentIp, err)
+		log.Printf("Failed to retrieve system metrics from agent at %s: %s", a.AgentIP, err)
 	}
 
-	containerMetricsLists, err := getContainerMetrics(agentIp, agentState, stats)
+	containerMetricsLists, err := a.getContainerMetrics(agentState, stats)
 	if err == nil {
 		for _, metricList := range containerMetricsLists {
 			recordsChan <- metricList
 		}
 	} else {
-		log.Printf("Failed to retrieve container metrics from agent at %s: %s", agentIp, err)
+		log.Printf("Failed to retrieve container metrics from agent at %s: %s", a.AgentIP, err)
 	}
 }
 
 // runs detect_ip => "10.0.3.26\n"
-func getAgentIp(stats chan<- StatsEvent) string {
+func (a *Agent) getIP(stats chan<- StatsEvent) error {
 	stats <- MakeEvent(AgentIpLookup)
-	cmdWithArgs := strings.Split(*agentIpCommandFlag, " ")
+	cmdWithArgs := strings.Split(a.IPCommand, " ")
 	ipBytes, err := exec.Command(cmdWithArgs[0], cmdWithArgs[1:]...).Output()
 	if err != nil {
 		stats <- MakeEvent(AgentIpLookupFailed)
-		log.Fatalf("Fetching Agent IP with -agent-ip-command='%s' failed: %s", *agentIpCommandFlag, err)
+		return err
 	}
 	ip := strings.TrimSpace(string(ipBytes))
 	if len(ip) == 0 {
 		stats <- MakeEvent(AgentIpLookupEmpty)
-		log.Fatalf("Agent IP fetched with -agent-ip-command='%s' is empty", *agentIpCommandFlag)
+		return err
 	}
-	//log.Printf("Agent IP obtained with -agent-ip-command='%s': %s\n", *agentIpCommandFlag, ip)
-	return ip
+
+	a.AgentIP = ip
+
+	return nil
 }
 
 // fetches container-level resource metrics from the agent (via /containers), emits to the framework topics (default 'metrics-<framework_id>')
-func getContainerMetrics(agentIp string, agentState *AgentState, stats chan<- StatsEvent) ([]*AvroDatum, error) {
-	rootJson, err := getJsonFromAgent(agentIp, "/containers", agentTestContainersFileFlag, stats)
+func (a *Agent) getContainerMetrics(agentState *AgentState, stats chan<- StatsEvent) ([]*AvroDatum, error) {
+	rootJson, err := a.getJsonFromAgent("/containers", agentTestContainersFileFlag, stats)
```

Comment:
See @rji comments above. There is a commit to update all the `Ip` -> `IP` and the like in the pipeline.